src/main/riddl/Events/eventContext.riddl

```
@@ -0,0 +1,197 @@
+                }
+                on command RescheduleEvent {
+                  then "set entity evenInfo.expectedStart to Event.start"
+                  and "set entity evenInfo.expectedEnd to Event.end"
+                  and "set entity lastModifiedBy to Event.reschedulingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command DelayEvent{
+                  then "morph Event entity to EventStatus.Delayed"
+                  and "set entity eventInfo.expectedStart to eventInfo.expectedStart + Event.duration"
+                  and "set entity eventInfo.expectedEnd to eventInfo.expectedEnd + Event.duration"
+                  and "set entity lastModifiedBy to Event.delayingMember"
+                  and "set entity lastModifiedOn to today()"
+                  // what about Event.reason? Do we want to store it somewhere?
+                }
+                on command StartEvent{
+                  then "morph Event entity to EventStatus.InProgress"
+                  and "set entity lastModifiedBy to Event.startingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command EndEvent{
+									then error "End has not started. Cannot end an event that has not started."
+                }
+                on command AddLiveUpdate{???}
+            }
+        }
+        state InProgress is {
+            fields{
+                eventId: EventId,
+                eventInfo: EventInfo,
+                eventMetaInfo: EventMetaInfo
+            }
+            handler InProgressEventHandler is {
+                on command ScheduleEvent{ ??? // is this allowed or does an event have to be delayed first?
+                }
+                on command ChangeEventInfo{ ??? // can info be changed while event is in progress? Would this be more likely to confuse or help users?
+                }
+                on command CancelEvent{
```

This command should throw an error when in the InProgress state.